### PR TITLE
Work towards Deprecating "melee_cut"

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
None

#### Purpose of change
Make all bonus damage follow the "melee_damage" array format.  I believe this will eliminate confusion in monster making by standardizing.  Standardizing should also help prepare the way for jsonized damage types in post 0.G.

#### Describe the solution
Well the capture wildcards find and replace failed but it gave me the idea of replacing everything in a way that allows me to easily go in and modify each line and I can then easily check when I've completed a given file.  But it's basically just find and replace.

#### Describe alternatives you've considered
Not doing this, but I think it's important.

#### Testing
Tests should continue to fail extensively until I've fixed everything I broke in the original find and replace :)

#### Additional context
This won't break third party mods but if `melee_cut` is phased out it will.

### CODE REVIEW

**Comment by Maleclypse** [comment]
I think I'll need to add the tropicataclysm folder to my project folders and do the same for their monsters.

**Comment by Maleclypse** [comment]
@dseguin You might know a way for me to do the find and replace with the old value for "melee_cut": '*', wildcarded so it replaces easily?

**Comment by NetSysFire** [comment]
If this is on Linux (or at least WSL) you can use sed:

```bash

sed 's/"melee_cut": \([0-9]\+\),/"melee_damage": [ { "damage_type": "cut", "amount": \1 } ],/g'
```

**Comment by LyleSY** [comment]
Is it/will it be possible then to change some of these damage types? Right now all dinos do bonus cut damage if they do bonus damage, but it would be better if the damage type matched the dino. For example a dino that rams or smashes should be bashing.

**Review by LyleSY — APPROVED** [review]
DinoMod stuff looks good. I'll add specific damage types in a followup PR, I'll try and pull in some vanilla ones too

**Comment by Maleclypse** [comment]
Need to understand if `melee_damage` is supported in `proportional` and `relative` and if not I will need to undo any of the changes to those two values and `melee_cut` won't be able to be fully deprecated yet.

File: data/json/monsters/fungus_zombie.json [hunk 1]
```diff
@@ -136,7 +136,8 @@
     "material": [ "bone" ],
     "description": "Fungi bloom from the crevices in the twisted bone growths that cover this creature.  Spores linger in the air around it.",
     "proportional": { "hp": 1.5, "speed": 1.5 },
-    "relative": { "melee_dice": 1, "melee_dice_sides": 5, "melee_cut": 2, "armor_bash": 4, "armor_cut": 6, "armor_bullet": 5 },
+    "relative": { "melee_dice": 1, "melee_dice_sides": 5, "armor_bash": 4, "armor_cut": 6, "armor_bullet": 5 },
```
Comment (Maleclypse) [inline data/json/monsters/fungus_zombie.json:139]:
```suggestion
    "relative": { "melee_dice": 1, "melee_dice_sides": 5, "armor_bash": 4, "melee_cut": 2, "armor_cut": 6, "armor_bullet": 5 },
```

File: data/json/monsters/fungus_zombie.json [hunk 2]
```diff
@@ -136,7 +136,8 @@
...
+    "melee_damage": [ { "damage_type": "cut", "amount": 2 } ],
```
Comment (Maleclypse) [inline data/json/monsters/fungus_zombie.json:140]:
```suggestion
```

File: data/mods/DinoMod/obsolete/monsters.json [hunk 3]
```diff
@@ -32,7 +32,7 @@
     "relative": {
       "melee_dice": 1,
       "melee_dice_sides": 5,
-      "melee_cut": 2,
+      "melee_damage": [ { "damage_type": "cut", "amount": 2 } ],
```
Comment (Maleclypse) [inline data/mods/DinoMod/obsolete/monsters.json:35]:
```suggestion
      "melee_cut": 2,
```

File: data/json/monsters/fungus_zombie.json [hunk 4]
```diff
@@ -115,7 +115,7 @@
     "relative": {
       "melee_dice": -1,
       "melee_dice_sides": 3,
-      "melee_cut": 4,
+      "melee_damage": [ { "damage_type": "cut", "amount": 4 } ],
```
Comment (Maleclypse) [inline data/json/monsters/fungus_zombie.json:118]:
```suggestion
      "melee_cut": 4,
```

File: data/json/monsters/zanimal_upgrade.json [hunk 5]
```diff
@@ -45,7 +45,7 @@
     "relative": {
       "melee_dice": 1,
       "melee_dice_sides": 5,
-      "melee_cut": 2,
+      "melee_damage": [ { "damage_type": "cut", "amount": 2 } ],
```
Comment (Maleclypse) [inline data/json/monsters/zanimal_upgrade.json:48]:
```suggestion
      "melee_cut": 2,
```

File: data/json/monsters/zanimal_upgrade.json [hunk 6]
```diff
@@ -122,7 +122,7 @@
...
```
Comment (Maleclypse) [inline data/json/monsters/zanimal_upgrade.json:125]:
```suggestion
      "melee_cut": 2,
```

File: data/json/monsters/zed_acid.json [hunk 7]
```diff
@@ -280,7 +280,7 @@
     "relative": {
       "melee_dice": 1,
       "melee_dice_sides": 4,
-      "melee_cut": 2,
+      "melee_damage": [ { "damage_type": "cut", "amount": 2 } ],
```
Comment (Maleclypse) [inline data/json/monsters/zed_acid.json:283]:
```suggestion
      "melee_cut": 2,
```

File: data/json/monsters/nether.json [hunk 8]
```diff
@@ -180,7 +180,7 @@
     "melee_skill": 5,
     "melee_dice": 2,
     "melee_dice_sides": 4,
-    "melee_cut": 22,
+    "melee_damage": [ { "damage_type": "cut", "amount": 22 } ],,
```
Comment (Maleclypse) [inline data/json/monsters/nether.json:183]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 22 } ],
```

File: data/json/monsters/mutant_mammal.json [hunk 9]
```diff
@@ -287,7 +287,7 @@
     "melee_skill": 8,
     "melee_dice": 3,
     "melee_dice_sides": 12,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/json/monsters/mutant_mammal.json:290]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/json/monsters/reptile_amphibian.json [hunk 10]
```diff
@@ -149,7 +149,7 @@
     "melee_skill": 4,
     "melee_dice": 1,
     "melee_dice_sides": 10,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/json/monsters/reptile_amphibian.json:152]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/json/monsters/yrax.json [hunk 11]
```diff
@@ -19,7 +19,7 @@
     "melee_skill": 10,
     "melee_dice": 3,
     "melee_dice_sides": 8,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/json/monsters/yrax.json:22]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/json/monsters/zed_ferrous.json [hunk 12]
```diff
@@ -166,7 +166,7 @@
     "melee_skill": 3,
     "melee_dice": 2,
     "melee_dice_sides": 6,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/json/monsters/zed_ferrous.json:169]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/json/monsters/zed_fusion.json [hunk 13]
```diff
@@ -307,7 +307,7 @@
     "melee_skill": 5,
     "melee_dice": 5,
     "melee_dice_sides": 2,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/json/monsters/zed_fusion.json:310]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/json/monsters/zed_skeletal.json [hunk 14]
```diff
@@ -141,7 +141,7 @@
     "melee_skill": 5,
     "melee_dice": 3,
     "melee_dice_sides": 6,
-    "melee_cut": 15,
+    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],,
```
Comment (Maleclypse) [inline data/json/monsters/zed_skeletal.json:144]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],
```

File: data/mods/Aftershock/mobs/aliens.json [hunk 15]
```diff
@@ -88,7 +88,7 @@
     "melee_dice": 2,
     "melee_dice_sides": 3,
     "attack_cost": 330,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/mods/Aftershock/mobs/aliens.json:91]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/Aftershock/mobs/robots.json [hunk 16]
```diff
@@ -239,7 +239,7 @@
     "melee_dice_sides": 4,
     "special_attacks": [ [ "SEARCHLIGHT", 1 ] ],
     "luminance": 60,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/mods/Aftershock/mobs/robots.json:242]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/CRT_EXPANSION/monsters/crt_monster.json [hunk 17]
```diff
@@ -64,7 +64,7 @@
     "melee_skill": 4,
     "melee_dice": 3,
     "melee_dice_sides": 3,
-    "melee_cut": 15,
+    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],,
```
Comment (Maleclypse) [inline data/mods/CRT_EXPANSION/monsters/crt_monster.json:67]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],
```

File: data/mods/CRT_EXPANSION/monsters/crt_monster.json [hunk 18]
```diff
@@ -108,7 +108,7 @@
     "melee_skill": 7,
     "melee_dice": 3,
     "melee_dice_sides": 6,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/mods/CRT_EXPANSION/monsters/crt_monster.json:111]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/Dark-Skies-Above/monsters/alien_fauna.json [hunk 19]
```diff
@@ -99,7 +99,7 @@
     "melee_skill": 4,
     "melee_dice": 2,
     "melee_dice_sides": 4,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/mods/Dark-Skies-Above/monsters/alien_fauna.json:102]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 20]
```diff
@@ -166,7 +166,7 @@
     "melee_skill": 6,
     "melee_dice": 4,
     "melee_dice_sides": 7,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/mods/DinoMod/monsters/dinosaur.json:169]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 21]
```diff
@@ -1434,7 +1434,7 @@
     "melee_skill": 8,
     "melee_dice": 2,
     "melee_dice_sides": 8,
-    "melee_cut": 15,
+    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],,
```
Comment (Maleclypse) [inline data/mods/DinoMod/monsters/dinosaur.json:1437]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],
```

File: data/mods/DinoMod/monsters/zed-dinosaur.json [hunk 22]
```diff
@@ -121,7 +121,7 @@
     "melee_skill": 4,
     "melee_dice": 4,
     "melee_dice_sides": 7,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/mods/DinoMod/monsters/zed-dinosaur.json:124]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/DinoMod/monsters/zed-dinosaur.json [hunk 23]
```diff
@@ -1066,7 +1066,7 @@
     "melee_skill": 5,
     "melee_dice": 2,
     "melee_dice_sides": 8,
-    "melee_cut": 15,
+    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],,
```
Comment (Maleclypse) [inline data/mods/DinoMod/monsters/zed-dinosaur.json:1069]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],
```

File: data/mods/Magiclysm/monsters/demon_spider.json [hunk 24]
```diff
@@ -79,7 +79,7 @@
     "melee_skill": 9,
     "melee_dice": 20,
     "melee_dice_sides": 8,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/monsters/demon_spider.json:82]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 25]
```diff
@@ -303,7 +303,7 @@
     "melee_skill": 5,
     "melee_dice": 5,
     "melee_dice_sides": 7,
-    "melee_cut": 12,
+    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/monsters/monsters.json:306]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 26]
```diff
@@ -559,7 +559,7 @@
     "melee_skill": 5,
     "melee_dice": 1,
     "melee_dice_sides": 1,
-    "melee_cut": 15,
+    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/monsters/monsters.json:562]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 15 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 27]
```diff
@@ -370,7 +370,7 @@
     "id": "mon_xe_unicorn",
     "type": "MONSTER",
     "name": { "str": "unicorn" },
-    "description": "A hooved grazing creature with a thick, shaggy, shimmering mane of hair, a sweeping tail, and powerful-looking muscles.  Impossible to confuse with a regular horse even if you don't spot the spiraled horn coming from the head.  Rumor has it that some of the Fair Lords have ways to tame and ride this magnificent creature.",
+    "description": "A hooved grazing creature with a thick, shaggy, shimmering mane of hair, a sweeping tail, and powerful-looking muscles.  Impossible to confuse with a regular horse even if you don't spot the spiraled horn coming from the head.  Rumor has it that some of the Fair Lord's have ways to tame and ride this magnificent creature.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:373]:
```suggestion
    "description": "A hooved grazing creature with a thick, shaggy, shimmering mane of hair, a sweeping tail, and powerful-looking muscles.  Impossible to confuse with a regular horse even if you don't spot the spiraled horn coming from the head.  Rumor has it that some of the Fair Lords have ways to tame and ride this magnificent creature.",
```

**Pull request merged**